### PR TITLE
Added more presenters in dependency injection modules

### LINKED ISSUE #1764: Add Koin for dependency management
- [ ] Move all instantiation of Retrofit in Dependency Graph

More tasks to be added

### PR DESCRIPTION
**Pull request by pallav12**
Fixes #1764 

Changes: added ChatPresenter, SettingsPresenter, LoginPresenter, SignupPresenter in dependency injection modules.
GIF
chatgif | signupgif
settingsgif | Signupsgif

### CODE REVIEW

File: app/src/main/java/org/fossasia/susi/ai/chat/ChatPresenter.kt [hunk 1]
```diff
@@ -78,7 +77,6 @@ class ChatPresenter(context: Context) :
     var queueExecuting = AtomicBoolean(false)
 
     override fun onAttach(chatView: IChatView) {
```
Comment (iamareebjamal) [inline app/src/main/java/org/fossasia/susi/ai/chat/ChatPresenter.kt]: Remove redundant functions

File: app/src/main/java/org/fossasia/susi/ai/chat/STTfragment.kt [hunk 2]
```diff
@@ -23,22 +23,23 @@ import org.fossasia.susi.ai.R
...
+import org.koin.core.parameter.parametersOf
 import timber.log.Timber
 
 /**
  * Created by meeera on 17/8/17.
  */
 class STTFragment : Fragment() {
     lateinit var recognizer: SpeechRecognizer
-    lateinit var chatPresenter: IChatPresenter
+    val chatPresenter: IChatPresenter by inject { parametersOf(this) }
```
Comment (iamareebjamal) [inline app/src/main/java/org/fossasia/susi/ai/chat/STTfragment.kt]: private

File: app/src/main/java/org/fossasia/susi/ai/login/LoginPresenter.kt [hunk 3]
```diff
@@ -196,7 +194,6 @@ class LoginPresenter(loginActivity: LoginActivity) :
     }
 
     override fun onDetach() {
-        loginView = null
```
Comment (iamareebjamal) [inline app/src/main/java/org/fossasia/susi/ai/login/LoginPresenter.kt:199]: same

File: app/src/main/java/org/fossasia/susi/ai/chat/ChatPresenter.kt (hunk 1 above)
Comment (pallav12) [reply app/src/main/java/org/fossasia/susi/ai/chat/ChatPresenter.kt]: Removed onAttach from every interface where it's redundant...

File: app/src/main/java/org/fossasia/susi/ai/chat/STTfragment.kt (hunk 2 above)
Comment (pallav12) [reply app/src/main/java/org/fossasia/susi/ai/chat/STTfragment.kt]: Okay

File: app/src/main/java/org/fossasia/susi/ai/login/LoginPresenter.kt (hunk 3 above)
Comment (pallav12) [reply app/src/main/java/org/fossasia/susi/ai/login/LoginPresenter.kt:199]: Okay...

**Comment by pallav12** [comment]
Updated everything according to review... I'll try to change lesser files from next PR

**Review by iamareebjamal — APPROVED** [review]

**Pull request merged**